File: 백준/Silver/BOJ_1912.java
package test;

import java.util.Scanner;

public class BOJ_1912 {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();

        int[] arr = new int[N];
        int[] dp = new int[N];

        for (int i = 0; i < N; i++) {
            arr[i] = sc.nextInt();
        }

        dp[0] = arr[0];
        int answer = dp[0];

        for (int i = 1; i < N; i++) {
            dp[i] = Math.max(arr[i], dp[i-1] + arr[i]);
            if (answer < dp[i]) {
                answer = dp[i];
            }
        }

        System.out.println(answer);

    }
}

File: 백준/Silver/BOJ_1912.py
MAX_ARR_SIZE = 100000
MIN_VALUE = -1000

N = int(input())
arr = list(map(int, input().split()))
dp = [MIN_VALUE] * MAX_ARR_SIZE
dp[0] = arr[0]

for i in range(1, N):
    dp[i] = max(arr[i], arr[i]+dp[i-1])

answer = max(dp)

print(answer)